homeassistant/components/tplink/climate.py

```diff
@@ -116,8 +116,8 @@ async def async_turn_off(self) -> None:
     @callback
     def _async_update_attrs(self) -> None:
         """Update the entity's attributes."""
-        self._attr_current_temperature = self._temp_feature.value
-        self._attr_target_temperature = self._target_feature.value
+        self._attr_current_temperature = cast(float | None, self._temp_feature.value)
+        self._attr_target_temperature = cast(float | None, self._target_feature.value)
```

Comment:
This is fine for now, but we should replace it with a TypeVar in the future so we don't have to do the casts